### ISSUE TITLE
[Bug Report] Attention masking is not used by model forward methods

### ISSUE THREAD

**Issue by jmsdao** [TransformerLensOrg/TransformerLens#467]
![image](https://github.com/neelnanda-io/TransformerLens/assets/30713160/d9011129-901d-4860-b3d1-778509d800ba)

**Comment by uralik** [comment]
IIUC the attention_mask is overwritten in the code if you don't set start_at_layer argument:

https://github.com/neelnanda-io/TransformerLens/blob/main/transformer_lens/HookedTransformer.py#L535-L546

this is also mentioned in the docstring:
https://github.com/neelnanda-io/TransformerLens/blob/main/transformer_lens/HookedTransformer.py#L511-L513

since it infers the pad attention mask from the tokens itself. In your case you dont have pad tokens, so the inferred attention mask has no effect.
